tests/e2e/e2e_setup_test.go

```diff
@@ -423,3 +452,108 @@ func noRestart(config *docker.HostConfig) {
 		Name: "no",
 	}
 }
+
+func (s *IntegrationTestSuite) writeGovProposals(c *chain) {
+	sendAmount := []sdk.Coin{
+		// 10photon
+		sdk.NewInt64Coin(photonDenom, 10),
+	}
+
+	type GovMessageSend struct {
+		Type   string     `json:"@type"`
+		From   string     `json:"from_address"`
+		To     string     `json:"to_address"`
+		Amount []sdk.Coin `json:"amount"`
+	}
+
+	msgSendMessages := []GovMessageSend{
+		GovMessageSend{
+			Type:   "/cosmos.bank.v1beta1.MsgSend",
+			From:   govModuleAddress,
+			To:     govSendMsgRecipientAddress,
+			Amount: sendAmount,
+		},
+	}
+
+	msgSendBody, err := json.MarshalIndent(struct {
+		Messages []GovMessageSend `json:"messages"`
+		Metadata string           `json:"metadata"`
+		Deposit  string           `json:"deposit"`
+	}{
+		Messages: msgSendMessages,
+		Metadata: "VGVzdGluZyAxLCAyLCAzIQ==",
```

Comment:
```suggestion
		Metadata:     b64.StdEncoding.EncodeToString([]byte("Testing 1, 2, 3!")),
```
or should this be `b64.URLEncoding.EncodeToString([]byte(data))` for URL Encoding?

Did you confirm that the metadata needs to be Base64 encoded before being submitted?

Reply:
When you run 
```shell
$gaiad tx gov submit-proposal --help

Submit a proposal along with some messages, metadata and deposit.
They should be defined in a JSON file.

Example:
$ gaiad tx gov submit-proposal path/to/proposal.json

Where proposal.json contains:

{
  // array of proto-JSON-encoded sdk.Msgs
  "messages": [
    {
      "@type": "/cosmos.bank.v1beta1.MsgSend",
      "from_address": "cosmos1...",
      "to_address": "cosmos1...",
      "amount":[{"denom": "stake","amount": "10"}]
    }
  ],
  "metadata: "4pIMOgIGx1vZGU=", // base64-encoded metadata
  "deposit": "10stake"
}

```

The cli docs suggest metadata should be base64 encoded, but there's no validation or enforcement. For the sake of compliance we could encode the original string in the test instead of hardcoding the encoded result.

Reply:
cool then let's keep it base64 encoded but update to include the code for encoding for clarity 👍 
the snippet i provided above should work, just need to add `b64` to the imports

Reply:
oh great you already did : )